src/Streamly/Internal/Data/Stream/StreamD.hs

```diff
@@ -3562,6 +3701,50 @@ scanlMx' :: Monad m
 scanlMx' fstep begin done s =
     (begin >>= \x -> x `seq` done x) `consM` postscanlMx' fstep begin done s
 
+data DualState s f t g = DualLeft s f | DualRight t g
+
+{-# INLINE_NORMAL postscanFold #-}
+postscanFold :: Monad m
+    => FL.Fold m a b -> Stream m a -> Stream m b
+postscanFold (FL.Fold fstep begin done) (Stream step state) =
+    Stream step' (DualLeft state begin)
+
+    where
+
+    {-# INLINE_LATE step' #-}
+    step' gst (DualLeft st acc) = do
+        r <- step (adaptState gst) st
+        case r of
+            Yield x s -> do
+                old <- acc
+                y <- fstep old x
+                case y of
+                    FL.Partial sres -> do
+                        v <- done sres
+                        v `seq`
+                            y `seq` return (Yield v (DualLeft s (return sres)))
+                    FL.Done bres ->
+                        return (Yield bres (DualRight s (return bres)))
+                    FL.Done1 bres ->
+                        return (Yield bres (DualRight s (return bres)))
+            Skip s -> return $ Skip (DualLeft s acc)
+            Stop -> return Stop
+    step' gst (DualRight st acc) = do
```

Comment:
`postScanFold` is like `scanOne`. I've written it with that assumption.

Reply:
The behaviour you are describing is of `scanMany`ish.

Reply:
I think as soon as the fold is done we need to terminate the scan, instead of draining the whole stream. I guess scanMany can also be useful. You can put a skeleton implementation for that and mark it unimplemented.